### ISSUE TITLE
AddRealm breaks attribute lookup

### ISSUE THREAD

**Issue by vladimir-mencl-eresearch** [eduvpn/vpn-user-portal#165]
Hi,

First, thanks for developing this great package - it was a pleasure to deploy it!

When experimenting with the LDAP settings (for talking to an AD server), I ran into a glitch.  I tried using the `addRealm` setting, but using it breaks further lookups for the same user (so looking up the ACL attribute for the user would break).

When `addRealm` is used, the code in LdapAuth.isValid() modifies `$authUser` to the form with the realm component added:
https://github.com/eduvpn/vpn-user-portal/blob/v2/src/LdapAuth.php#L72

However, the `$userFilter` used for further lookups end up using this modified value of `$authUser`:
https://github.com/eduvpn/vpn-user-portal/blob/v2/src/LdapAuth.php#L89
- which does NOT work with search filters like `'(sAMAccountName={{UID}})'` - this should be matched against the original value of `$authUser`.

Please let me know whether you'd welcome a Pull-Request fixing this - or whether you'd rather fix this internally with your prefferred coding style.

Cheers,
Vlad

**Comment by fkooman** [comment]
Hi Vlad! This is actually the way it is supposed to work... When using `addRealm` we assume you are using `userPrincipalName` with AD. Only then it makes sense to use the `addRealm` option. Obviously for `sAMAccountName` this won't work. See the last section of the example configuration file here: https://github.com/eduvpn/documentation/blob/v2/LDAP.md#configuration

Could be I am missing something though?

**Comment by vladimir-mencl-eresearch** [comment]
Hi,

Thanks for the reply!

I see I just made a huge oversight on the Active Directory side - I always used `sAMAccountName` and this is the first time I'm hearing about `userPrincipalName`.  Well, one's always learning...

And I went primarily by the stock `/etc/vpn-user-portal/config.php` I got with the package - only looking up things to add from https://github.com/eduvpn/documentation/blob/v2/LDAP.md#configuration, but I missed that with addRealm, one needs to use a different `userFilterTemplate`.

So I expect (can't easily test, as the server is already in production) this would work - but perhaps it could be highlighted in the documentation?  Though, perhaps, one just has to read properly ... and not just scanned as I did.

Good with me either way - either with a doc change or leaving as-is.

Thanks for the patience with me - and thanks again for creating this great package!

Cheers,
Vlad
